hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/lock/OzoneManagerLock.java

```diff
@@ -514,29 +615,56 @@ public void cleanup() {
     // Name of the resource.
     private String name;
 
-    // This helps in maintaining lock related variables locally confined to a
-    // given thread.
+    // This helps in maintaining read lock related variables locally confined
+    // to a given thread.
     private final ThreadLocal<LockUsageInfo> readLockTimeStampNanos =
         ThreadLocal.withInitial(LockUsageInfo::new);
 
+    // This helps in maintaining write lock related variables locally confined
+    // to a given thread.
+    private final ThreadLocal<LockUsageInfo> writeLockTimeStampNanos =
+        ThreadLocal.withInitial(LockUsageInfo::new);
+
     /**
-     * Sets the time (ns) when the lock holding period begins specific to a
+     * Sets the time (ns) when the read lock holding period begins specific to a
      * thread.
      *
-     * @param startHeldTimeNanos lock held start time (ns)
+     * @param startReadHeldTimeNanos read lock held start time (ns)
      */
-    void setStartHeldTimeNanos(long startHeldTimeNanos) {
-      readLockTimeStampNanos.get().setStartHeldTimeNanos(startHeldTimeNanos);
+    void setStartReadHeldTimeNanos(long startReadHeldTimeNanos) {
+      readLockTimeStampNanos.get()
+          .setStartReadHeldTimeNanos(startReadHeldTimeNanos);
     }
 
     /**
-     * Returns the time (ns) when the lock holding period began specific to a
-     * thread.
+     * Sets the time (ns) when the write lock holding period begins specific to
+     * a thread.
+     *
+     * @param startWriteHeldTimeNanos write lock held start time (ns)
+     */
+    void setStartWriteHeldTimeNanos(long startWriteHeldTimeNanos) {
+      writeLockTimeStampNanos.get()
+          .setStartWriteHeldTimeNanos(startWriteHeldTimeNanos);
+    }
+
+    /**
+     * Returns the time (ns) when the read lock holding period began specific to
+     * a thread.
+     *
+     * @return read lock held start time (ns)
+     */
+    long getStartReadHeldTimeNanos() {
+      return readLockTimeStampNanos.get().getStartReadHeldTimeNanos();
```

Comment:
I think you can remove the element for safer side.

`readLockTimeStampNanos.remove();`